hcl2template/types.datasource.go

```diff
@@ -11,8 +11,8 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
-// DataBlock references an HCL 'data' block.
-type Datasource struct {
+// DatasourceBlock references an HCL 'data' block.
+type DatasourceBlock struct {
```

Comment:
I renamed this one to be consistent with other block types ( also I needed to create that new `Datasource ComponentKind` )